src/run.ts

```diff
@@ -21,7 +25,9 @@ export default async function run(): Promise<void> {
   try {
     const inputs = getInputs()
     const artifacts: Artifact[] = inputs.artifacts.map(artifact => ({
-      ...artifact,
+      path: artifact.path,
+      kind: artifact.kind,
+      filename: artifact.save_as || artifact.path.split('/').pop() || 'file',
```

Comment:
From what I can tell `artifact.path.split('/').pop()` will always return a string since `artifact.path` must be present, but I think since `pop` can return `undefined` I was running into a typescript error here and had to append `|| 'file'`. Let me know if anybody with better knowledge of typescript has any ideas.

Reply:
I don't think you'll be able to get around this. Technically you could tell TS that String.prototype.split returns `[string, ...string[]]` and then maybe it would work? But that would be way more unnatural than just falling back to 'file'.